File: load_datasets.py
#!pip install gdown pandas numpy matplotlib seaborn scikit-learn tensorflow keras-tuner

import pandas as pd
import gdown
import os
# Google Drive file IDs
FILE_IDS = {
    "train": "1PT-8M-Nbw8aBGXXdonN7QCe59AqEBYOs",      
    "validation": "1o_211YxzPtY2_Gz2d9xRfOALu7AjPzF_",
    "test": "1y0OcL5_S0PtJW2aeUq5Qm7cy6ehdfuob"
}

def download_and_load_csv(file_id, name):
    filename = f"{name}_dataset.csv"

    if os.path.exists(filename):
        print(f"{filename} already exists - loading from disk")
    else:
        print(f"Downloading {name} dataset from Google Drive...")
        url = f"https://drive.google.com/uc?id={file_id}"
        try:
            gdown.download(url, filename, quiet=False)
        except Exception as e:
            raise ConnectionError(f"Failed to download {name} dataset: {str(e)}")

    try:
        return pd.read_csv(filename)
    except Exception as e:
        raise ValueError(f"Could not load {filename} as CSV: {str(e)}")

def load_all_datasets():
    """Load all datasets from Google Drive or local cache"""
    datasets = {}
    for name, file_id in FILE_IDS.items():
        datasets[name] = download_and_load_csv(file_id, name)
    return datasets

File: load_datasets.R
# Install once if needed:
install.packages("googledrive")
install.packages("readr")

library(googledrive)
library(readr)


# --- load_datasets.R ---

library(googledrive)
library(readr)

# Google Drive file IDs (same mapping as your Python code)
files <- list(
  training   = "1PT-8M-Nbw8aBGXXdonN7QCe59AqEBYOs",
  test       = "1y0OcL5_S0PtJW2aeUq5Qm7cy6ehdfuob",
  validation = "1o_211YxzPtY2_Gz2d9xRfOALu7AjPzF_"
)

download_and_load_csv <- function(file_id, name) {
  filename <- paste0(name, "_dataset.csv")
  
  # Download if file does not exist
  if (!file.exists(filename)) {
    message(sprintf("⬇ Downloading %s dataset from Google Drive...", name))
    drive_download(as_id(file_id), path = filename, overwrite = TRUE)
  } else {
    message(sprintf("%s already exists.", filename))
  }
  
  # Read CSV
  tryCatch({
    read_csv(filename, show_col_types = FALSE)
  }, error = function(e) {
    stop(sprintf("Could not load %s as CSV: %s", filename, e$message))
  })
}

load_all_datasets <- function() {
  datasets <- list()
  for (name in names(files)) {
    datasets[[name]] <- download_and_load_csv(files[[name]], name)
  }
  datasets
}
